pallets/msa/src/benchmarking.rs

```diff
@@ -101,30 +101,34 @@ benchmarks! {
 	}: _ (RawOrigin::Signed(caller.clone()))
 	verify {
 		assert!(Msa::<T>::get_msa_by_public_key(caller).is_some());
+		assert_eq!(frame_system::Pallet::<T>::events().len(), 1);
 	}
 
 	create_sponsored_account_with_delegation {
+		let s in 0 .. T::MaxSchemaGrantsPerDelegation::get();
 
 		let caller: T::AccountId = whitelisted_caller();
 		assert_ok!(Msa::<T>::create(RawOrigin::Signed(caller.clone()).into()));
 		assert_ok!(Msa::<T>::create_provider(RawOrigin::Signed(caller.clone()).into(),Vec::from("Foo")));
 
-		let (payload, signature, key) = create_payload_and_signature::<T>();
+		let schemas: Vec<SchemaId> = (0 .. s as u16).collect();
+		T::SchemaValidator::set_schema_count(schemas.len().try_into().unwrap());
+		let (payload, signature, key) = create_payload_and_signature::<T>(schemas);
 
 	}: _ (RawOrigin::Signed(caller), key, signature, payload)
+	verify {
+		assert_eq!(frame_system::Pallet::<T>::events().len(), 2);
+	}
 
 	revoke_delegation_by_provider {
-		let s in 5 .. 1005;
```

Comment:
why are you removing this> It seems like there was a relationship between number of delegations and removing weight

Reply:
Not easy to know what total delegations there are it is not feasible to get accurate numbers.